File: Sorting/insertionSort.c
// Implementation of Insertion Sort
#include<stdio.h>

void input(int A[],int n)

{

int i;
 
printf("Enter %d number of elements: ",n); for(i=0;i<n;i++)
scanf("%d",&A[i]);

}

void display(int A[],int n)

{

int i; for(i=0;i<n;i++)
printf("%d ",A[i]);

}

void insertionSort(int arr[],int n)

{

int i,j;

for(int i=1;i<n;i++)

{

int key=arr[i]; int j= i-1;
while(j>=0 && arr[j]>key)

{

arr[j+1]=arr[j]; j=j-1;
}

arr[j+1]=key;

}

}

void main()

{

int A[20],n;

printf("\nEnter the size of the array:");
 
scanf("%d",&n); input(A,n);
printf("\nArray before sorting:"); display(A,n);
insertionSort(A,n); printf("\nArray after sorting:"); display(A,n);
}

/* Output:
Enter the size of the array:5

Enter 5 number of elements: 6 4 8 5 7

Array before sorting:6 4 8 5 7

Array after sorting:4 5 6 7 8 */

File: Kotlin/InsertionSort.kt
fun insertionSort(arr: IntArray) {
    val lastIndex: Int = arr.size - 1

    for (i in 1..lastIndex) {
        val temp: Int = arr[i]
        var holePosition: Int = i
        while(holePosition > 0 && arr[holePosition - 1] > temp) {
            arr[holePosition] = arr[holePosition - 1]
            holePosition--
        }
        arr[holePosition] = temp
    }
}

fun printArray(arr: IntArray) {
    val lastIndex: Int = arr.size - 1;
    for (i in 0..lastIndex) {
        val num = arr[i]
            print("$num ")
    }
    println("")
}

fun main(args: Array<String>) {
    val arr: IntArray = intArrayOf(15, 3, 12, 6, -9, 9, 0);
    print("Before Sorting: ")
    printArray(arr)

    insertionSort(arr)
    print("After Sorting: ")
    printArray(arr)
}
